math/src/main/java/com/powsybl/math/timeseries/CompactStringBuffer.java

```diff
@@ -18,17 +19,13 @@
 
     private final BiList<String> dict = new BiList<>();
 
-    public CompactStringBuffer(int size) {
-        this.buffer = createIntBuffer(size);
+    public CompactStringBuffer(IntFunction<ByteBuffer> byteBufferAllocator, int size) {
+        this.buffer = byteBufferAllocator.apply(size * Integer.BYTES).asIntBuffer();
```

Comment:
Assert that byteBufferAllocator is not null

Reply:
Done